nbqa/__main__.py

```diff
@@ -451,6 +455,73 @@ def _run_command(
     return out, err, output_code, mutated
 
 
+def _find_config_file(
+    command: str,
+) -> Tuple[Optional[str], Optional[ConfigParser], Optional[str]]:
+    """
+    Find config file.
+
+    Parameters
+    ----------
+    command
+        Third-party tool being run.
+
+    Returns
+    -------
+    config_file
+        Name of config file.
+    config
+        ConfigParser object which has read in config file.
+    config_prefix
+        Prefix of sections in config file.
+    """
+    config_files = CONFIG_FILES
+    for config_file in config_files:
+        if config_file == "pyproject.toml":
+            continue  # Will be supported in a future PR.
+        config = configparser.ConfigParser()
+        config.read(config_file)
+        for section in NBQA_CONFIG_SECTION:
+            if config.has_section(f"nbqa.{section}"):
+                return config_file, config, "nbqa."
+    config = configparser.ConfigParser()
+    config.read(".nbqa.ini")
+    if config.has_section(command):
+        return ".nbqa.ini", config, ""
+    return None, None, None
+
+
+def _get_option(
+    config_file: str, config: ConfigParser, section: str, option: str
+) -> Optional[str]:
+    """
+    Get option from config file.
+
+    Parameters
+    ----------
+    config_file
+        Name of config file.
+    config
+        ConfigParser object which has read in config file.
+    section
+        e.g. config, mutate, ...
+    option
+        e.g. flake8, black, ...
+
+    Returns
+    -------
+    str
+        Parsed option.
+    """
+    if config_file != ".nbqa.ini":
```

Comment:
Querying from different configurations `nbqa.ini`, `setup.cfg`, `pyproject.toml` seems to be a good place to apply polymorphism, Different configuration handlers exposing same interface to query the config. We can avoid these if/else structures. @MarcoGorelli Your thoughts on this? Do you have some sort of preference in using classes?